src/progress.rs

```diff
@@ -40,6 +42,7 @@ enum ProgressDrawTargetKind {
     Term(Term, Option<ProgressDrawState>, Option<Duration>),
     Remote(usize, Mutex<Sender<(usize, ProgressDrawState)>>),
     Hidden,
+    Text(bool),
```

Comment:
Could this `bool` be perhaps replaced with a `Box<dyn io::Write>` along with a new method like `pub fn to_writer(writer: Box<...>)` to allow for more uses?